go-controller/pkg/clustermanager/subnetallocator/host_subnet_allocator_test.go

```diff
@@ -258,153 +277,49 @@ func TestController_allocateNodeSubnets_ReleaseOnError(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	sna := NewHostSubnetAllocator()
-	if err := sna.InitRanges(ranges); err != nil {
+	config.Default.ClusterSubnets = ranges
+
+	netInfo, err := util.NewNetInfo(
+		&ovncnitypes.NetConf{
+			NetConf: cnitypes.NetConf{Name: types.DefaultNetworkName},
+		},
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	sna := &HostSubnetAllocator{
+		netInfo:                netInfo,
+		clusterSubnetAllocator: NewSubnetAllocator(),
+	}
+
+	if err := sna.InitRanges(); err != nil {
 		t.Fatalf("Failed to initialize network ranges: %v", err)
 	}
 
 	// Mark all v6 subnets already allocated to force an error in AllocateNodeSubnets()
-	if err := sna.MarkSubnetsAllocated("blah", ovntest.MustParseIPNet("2000::/127")); err != nil {
-		t.Fatalf("MarkSubnetsAllocated() expected no error but got: %v", err)
+	if err := sna.clusterSubnetAllocator.MarkAllocatedNetworks("blah", ovntest.MustParseIPNet("2000::/127")); err != nil {
+		t.Fatalf("MarkAllocatedNetworks() expected no error but got: %v", err)
 	}
 
 	// test network allocation works correctly
-	_, v4usedBefore, _, v6usedBefore := sna.base.Usage()
-	got, allocated, err := sna.AllocateNodeSubnets("testNode", nil, true, true)
+	_, v4usedBefore, _, v6usedBefore := sna.clusterSubnetAllocator.Usage()
+	got, allocated, err := sna.allocateNodeSubnets(sna.clusterSubnetAllocator, "testNode", nil, true, true)
 	if err == nil {
-		t.Fatalf("AllocateNodeSubnets() expected error but got success")
+		t.Fatalf("allocateNodeSubnets() expected error but got success")
 	}
 	if got != nil {
-		t.Fatalf("AllocateNodeSubnets() expected no existing host subnets, got %v", got)
+		t.Fatalf("allocateNodeSubnets() expected no existing host subnets, got %v", got)
 	}
 	if allocated != nil {
-		t.Fatalf("AllocateNodeSubnets() expected no allocated subnets, got %v", allocated)
+		t.Fatalf("allocateNodeSubnets() expected no allocated subnets, got %v", allocated)
 	}
 
-	_, v4usedAfter, _, v6usedAfter := sna.base.Usage()
+	_, v4usedAfter, _, v6usedAfter := sna.clusterSubnetAllocator.Usage()
 	if v4usedAfter != v4usedBefore {
 		t.Fatalf("Expected %d v4 allocated subnets, but got %d", v4usedBefore, v4usedAfter)
 	}
 	if v6usedAfter != v6usedBefore {
 		t.Fatalf("Expected %d v6 allocated subnets, but got %d", v6usedBefore, v6usedAfter)
 	}
 }
-
-func ipnetStringsToSlice(strings []string) ([]*net.IPNet, error) {
-	slice := make([]*net.IPNet, 0, len(strings))
-	for _, s := range strings {
-		_, subnet, err := net.ParseCIDR(s)
-		if err != nil {
-			return nil, fmt.Errorf("error parsing subnet %s", s)
-		}
-		slice = append(slice, subnet)
-	}
-	return slice, nil
-}
-
-func TestController_markSubnetsAllocated(t *testing.T) {
```

Comment:
Not needed since I removed the `markSubnetsAllocated` which was just a passthrough to SubnetAllocator `MarkAllocatedNetworks` which is already being tested by itself

Reply:
tks for explaining.